### PR TITLE
oom

### PR DESCRIPTION
**Pull request by vince-hz**
- Add `BridgeCallRecorder`
- WhiteboardView recording and reload
- Add bridge call command type
- Refactor apple pencil prepare
- Add more callbacks for sdk delegate
- Recover from web crash
- Add crash test
- CHANGELOG

### CODE REVIEW

File: Whiteboard/Classes/SDK/BridgeCallRecorder.m [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+//  Whiteboard
+//
+//  Created by xuyunshi on 2022/7/21.
+//
+
+#import "BridgeCallRecorder.h"
+
+@interface BridgeCallRecorder ()
+
+@property (nonatomic, strong) NSMutableArray<WhiteCallBridgeCommand *>* recordedCommands;
```
Comment (leavesster) [inline Whiteboard/Classes/SDK/BridgeCallRecorder.m:12]: 叫 recordCommands 会好一些

File: Whiteboard/Classes/SDK/BridgeCallRecorder.m [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+@property (nonatomic, copy) NSDictionary<NSString *, NSNumber *>*recordingKeys;
```
Comment (leavesster) [inline Whiteboard/Classes/SDK/BridgeCallRecorder.m:13]: record Key 即可

File: Whiteboard/Classes/SDK/BridgeCallRecorder.m [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        WhiteCallBridgeCommand *command = _lockedArray[index];
+        __weak typeof(self) weakSelf = self;
+        __weak typeof(view) weakWebView = view;
+        [self resumeCommandFromBridgeView:view command:command completionHandler:^{
+            __strong typeof(weakSelf) strongSelf = weakSelf;
+            strongSelf->_leftLockedArrayResumeCount -= 1;
+            [strongSelf loopLockedCommandsFromBridgeView:weakWebView completionHandler:completionHandler];
+        }];
+    } else {
+        completionHandler();
```
Comment (leavesster) [inline Whiteboard/Classes/SDK/BridgeCallRecorder.m:58]: 如果 completionHandler 是  nullable，应该要判空一下。后面的也是一样。

**Comment by vince-hz** [comment]
修改`_lockedArray` ,  `_leftLockedArrayResumeCount` 为 property， 评论被误删了

**Review by leavesster — APPROVED** [review]

**Pull request merged**